### ISSUE TITLE
Supplying renderer prop to Stage yields "Uncaught TypeError: Cannot read property 'nodeType' of undefined"

### ISSUE THREAD

**Issue by michalochman** [Izzimach/react-pixi#89]
The following code is working:
```javascript
var PIXI = require('pixi.js')
var React = require('react')
var ReactPIXI = require('react-pixi')

function init() {
  ReactPIXI.render(
    React.createElement(ReactPIXI.Stage, {
      width: 400,
      height: 300,
    }, [
      React.createElement(ReactPIXI.Text, {
        key: 'react-pixi-text-1',
        text: 'react-pixi-test'
      })
    ]),
    document.getElementById('app')
  )
}

window.addEventListener('load', init, false)
```

but as soon as you try to use custom renderer:
```diff
var PIXI = require('pixi.js')
var React = require('react')
var ReactPIXI = require('react-pixi')

function init() {
+ const renderer = PIXI.autoDetectRenderer({
+   width: 400,
+   height: 300,
+ })
+
  ReactPIXI.render(
    React.createElement(ReactPIXI.Stage, {
      width: 400,
      height: 300,
+     renderer: renderer,
    }, [
      React.createElement(ReactPIXI.Text, {
        key: 'react-pixi-text-1',
        text: 'react-pixi-test'
      })
    ]),
    document.getElementById('app')
  )
}

window.addEventListener('load', init, false)
```

throws:
```
Uncaught TypeError: Cannot read property 'nodeType' of undefined
    at Function.insertTreeBefore (DOMLazyTree.js:59)
    at Object._mountImageIntoNode (ReactMount.js:518)
    at mountComponentIntoNode (ReactMount.js:112)
    at ReactReconcileTransaction.perform (Transaction.js:143)
    at batchedMountComponentIntoNode (ReactMount.js:126)
    at ReactDefaultBatchingStrategyTransaction.perform (Transaction.js:143)
    at Object.batchedUpdates (ReactDefaultBatchingStrategy.js:62)
    at Object.batchedUpdates (ReactUpdates.js:97)
    at Object._renderNewRootComponent (ReactMount.js:319)
    at Object._renderSubtreeIntoContainer (ReactMount.js:401)

| (anonymous) | @ | DOMLazyTree.js:59
| _mountImageIntoNode | @ | ReactMount.js:518
| mountComponentIntoNode | @ | ReactMount.js:112
| perform | @ | Transaction.js:143
| batchedMountComponentIntoNode | @ | ReactMount.js:126
| perform | @ | Transaction.js:143
| batchedUpdates | @ | ReactDefaultBatchingStrategy.js:62
| batchedUpdates | @ | ReactUpdates.js:97
| _renderNewRootComponent | @ | ReactMount.js:319
| _renderSubtreeIntoContainer | @ | ReactMount.js:401
| render | @ | ReactMount.js:422
| init | @ | index.js:11
```

I am using:
- React v15.6.1
- react-pixi v0.9.12
- pixi.js v4.5.2

**Comment by michalochman** [comment]
In react-pixi v0.9.8 it worked with the following changes:

```diff
var PIXI = require('pixi.js')
var React = require('react')
var ReactPIXI = require('react-pixi')

+class Stage extends React.Component {
+  componentDidMount() {
+    document.getElementById('app').appendChild(this.props.renderer.view)
+  }
+
+  componentWillUnmount() {
+    document.getElementById('app').removeChild(this.props.renderer.view)
+  }
+
+  render() {
+    return (
+      <ReactPIXI.Stage {...this.props} />
+    )
+  }
+}
+
function init() {
  const renderer = PIXI.autoDetectRenderer({
    width: 400,
    height: 300,
  })

  ReactPIXI.render(
-   React.createElement(ReactPIXI.Stage, {
+   React.createElement(Stage, {
      height: 400,
      width: 300,
      renderer: renderer,
    }, [
      React.createElement(ReactPIXI.Text, {
        key: 'react-pixi-text-1',
        style: {
          fill: 'red',
        },
        text: 'react-pixi-test'
      })
    ]),
    document.getElementById('app')
  )
}

window.addEventListener('load', init, false)
```

However, after updating to react-pixi v0.9.9, it throws the error mentioned above.

**Comment by Izzimach** [comment]
Yeah, when you use a custom renderer the lib won't create/manage the canvas for you. This was originally used for some framework with a hard-coded canvas (Ejecta, maybe?)

This should probably be an example and/or test since it will probably break again. I can probably just copy/paste the code above for that.

**Comment by michalochman** [comment]
Do you think it can be fixed to work again? It worked great in v0.9.8.

This basically is a workaround I'm using to have an instance of renderer that can be later used by other components. I would not need to manage my own custom renderer if ReactPIXI.Stage exposed the renderer it manages to the children somehow (through a context API maybe?).
